givitsite/friendreq/migrations/createsuperuser.py

```diff
@@ -12,9 +10,9 @@ class Migration(migrations.Migration):
     ]
 
     def generate_superuser(apps, schema_editor):
```

Comment:
I don't thing the new code produce any different results then the old code - only now you've removed your ability to customize those values for example in production...

Reply:
you are right.
when we run -vagrant up. the app is working on the background. 
I couldn't figure out how can we change those variables. 
now, (as written in the README) to create  a new superuser you have to do it using the command line

Reply:
It is a really small change, I didn't want to have a code that I don't fully understand (how to use it)

Reply:
![image](https://user-images.githubusercontent.com/61931613/92592237-ef82ac00-f2a7-11ea-9219-c0bf2dde5688.png)
this is the part where it is explained (in the README)

Reply:
Since the code in question is in a migration file - to make it work you need to set the environment variables _before_ the migration runs - the use case for this is when you launch the app for the fist time in a production environment with an empty database.

Once the migration had run, it will never run again unless you either roll it back or completely remove the database and start from scratch.

Generally you can't use this in vagrant because by the time you can do `vagrant ssh` the migrations  had already ran.

This is not for changing the user/password "on the fly" so to speak, for that you can indeed use `createsuperuser`. Though as with other systems, I'd generally just avoid using the super user on a regular basis and create other user accounts instead.

Reply:
Even when I do vagrant destroy, How can I change/ set the environment variables on this process?

I probably miss something but my logic goes like this:
1. the environment doesn't exist before I do vagrant up.
2. Since we want the app to start automatically, "migrate" is one command in the bootstrap.sh

Assuming that the admin wants to have other use/pass (security...) how can he do it?

Reply:
In production you do not use Vagrant.

Vagrant is only for local laptop _development_.

When you, for example, run the app in a Kubernetes POD in production, you would set the environment variables on the POD (Or more likely, configure a Kubernetes secret that will get linked into the POD as variables).

Reply:
correct me if I'm wrong, in the code that you showed us when we all migrate the function "generate_superuser()" looks for the environment variables:

- 'DJANGO_SU_NAME'

- 'DJANGO_SU_EMAIL'

- 'DJANGO_SU_PASSWORD'

and only is it doesn't find them the default values will be defined

Reply:
understood!
I'll cancel the PR and write the README again

Reply:
yes. how does that change what I've said? for any of that to happen you need:

1. The function  to run, which means the particular migration it is a part of needs to run
2. It needs to run in an environment where the variables are set.

Unless you change your bootstrap script, you can't set the variables in the Vagrant environment early enough, but that is find because this is not for Vagrant.

Reply:
I still having trouble seeing how it will look at production...

I assume that we will discuss it on one of our sessions

Reply:
next one, and the one after that as a matter of fact :)